python/src/iceberg/types.py

```diff
@@ -181,29 +275,33 @@ class ListType(IcebergType):
         ListType(element_id=3, element_type=StringType(), element_required=True)
     """
 
-    element_id: int = field()
-    element_type: IcebergType = field()
-    element_required: bool = field(default=True)
-    element: NestedField = field(init=False, repr=False)
-
-    def __post_init__(self):
-        object.__setattr__(
-            self,
-            "element",
-            NestedField(
-                name="element",
-                required=self.element_required,
-                field_id=self.element_id,
-                field_type=self.element_type,
-            ),
+    class Config:
+        fields = {"element_field": {"exclude": True}}
```

Comment:
Good question! This is because `PrivateField` requires the variable name to start with an underscore, and these variables shouldn't be used outside of the class itself. Because we use the `element_field` in the visitor as well, it makes more sense to make it an actual field, and omit it when we serialize the `ListType`.

Reply:
Sounds reasonable to me.